### PR TITLE
Add custom GraphQL enum type to fix #1007

### LINKED ISSUE #1007: Using an `Enum` as default value in resolver method raises `GraphQLError`
I tried to use an `Enum` as default argument for a resolver method:

```python
@strawberry.enum
class VersionSequence(Enum):
    FULL = "full"
    SHORT = "short"
    PATCH = "patch"

@strawberry.type
class Info:
    @strawberry.field
    def version(
        self, sequence: VersionSequence = VersionSequence.FULL
    ) -> str:
        return getVersion(sequence.value)
```
When accessing the GraphQL endpoint the following error is raised:
```python
Enum 'VersionSequence' cannot represent value: <VersionSequence instance>
Traceback (most recent call last):
  File "/home/mawiegand/.cache/pypoetry/virtualenvs/api-LQwpF9HC-py3.9/lib/python3.9/site-packages/graphql/execution/execute.py", line 617, in resolve_field
    result = resolve_fn(source, info, **args)
  File "/home/mawiegand/.cache/pypoetry/virtualenvs/api-LQwpF9HC-py3.9/lib/python3.9/site-packages/strawberry/middleware.py", line 29, in resolve
    result = next_(root, info, **kwargs)
  File "/home/mawiegand/.cache/pypoetry/virtualenvs/api-LQwpF9HC-py3.9/lib/python3.9/site-packages/graphql/type/introspection.py", line 454, in default_value
    value_ast = ast_from_value(item[1].default_value, item[1].type)
  File "/home/mawiegand/.cache/pypoetry/virtualenvs/api-LQwpF9HC-py3.9/lib/python3.9/site-packages/graphql/utilities/ast_from_value.py", line 63, in ast_from_value
    ast_value = ast_from_value(value, type_.of_type)
  File "/home/mawiegand/.cache/pypoetry/virtualenvs/api-LQwpF9HC-py3.9/lib/python3.9/site-packages/graphql/utilities/ast_from_value.py", line 108, in ast_from_value
    serialized = type_.serialize(value)  # type: ignore
  File "/home/mawiegand/.cache/pypoetry/virtualenvs/api-LQwpF9HC-py3.9/lib/python3.9/site-packages/graphql/type/definition.py", line 1155, in serialize
    raise GraphQLError(
graphql.error.graphql_error.GraphQLError: Enum 'VersionSequence' cannot represent value: <VersionSequence instance>
```
Querying without arguments seems to be working. Probably only the schema representation is affected.

Environment information:
Strawberry: `0.65.3`
Python: `[3.7, 3.9]`

### PR DESCRIPTION
**Pull request by jkimbo**
## Description

This PR fixes the issue with defining an enum member as the default value for an argument by subclassing the GraphQLEnumType from graphql-core and overriding the serialise function. This has the happy side effect of removing the need for the conversion layer `_convert_enums_to_values` in the field resolvers.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #1007

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)

File: strawberry/schema/schema_converter.py [hunk 1]
```diff
@@ -33,6 +34,13 @@
 from .types.scalar import get_scalar_type
 
 
+class CustomGraphQLEnumType(GraphQLEnumType):
```
Comment (DoctorJohn) [inline strawberry/schema/schema_converter.py:37]: Clean solution 👍

Should we add a comment documenting why this custom class was added? The old code had the following comment:

```python
# graphql-core expects a resolver for an Enum type to return
# the enum's *value* (not its name or an instance of the enum).
```

**Review by DoctorJohn — APPROVED** [review]

File: strawberry/schema/schema_converter.py (hunk 1 above)
Comment (jkimbo) [reply strawberry/schema/schema_converter.py:37]: Good point, will add.

File: strawberry/field.py [hunk 2]
```diff
@@ -272,18 +254,17 @@ def _resolver(_source: Any, info: GraphQLResolveInfo, **kwargs):
...
-                        yield _convert_enums_to_values(self, value)
+                        yield value
 
                 return yield_results(result)
 
             if iscoroutine(result):  # pragma: no cover
 
                 async def await_result(result):
-                    return _convert_enums_to_values(self, await result)
+                    return await result
```
Comment (patrick91) [inline strawberry/field.py:264]: so much better 💯

**Review by patrick91 — APPROVED** [review]
This is great! Feel free to merge it any time!

**Pull request merged**